src/pynxtools_em/concepts/mapping_functors_pint.py

```
@@ -231,25 +305,22 @@ def map_functor(
     prfx_trg: str,
     ids: list,
     template: dict,
-    **kwargs,
+    trg_dtype_key: str = "",
 ) -> dict:
-    if "trg_dtype_key" in kwargs:
-        trg_dtype = kwargs["trg_dtype"]
-        # this will force the conversion of all instance data from src to match trg_dtype
     for cmd in cmds:
         case = get_case(cmd)
         if case == "case_one":  # str
```

consider making this a lookup table that calls individual functions:
```
case_handler = {
    "case_one": handle_case_one,
    "case_two_str": handle_case_two_str,
}
```
And then calling `case_handler[case]()`. This makes it more maintainable. You can also raise an error if the case doesn't match.

---

Currently such refactoring is not planned